src/cowboy_swagger.erl

```diff
@@ -159,6 +158,62 @@ dec_json(Data) ->
       throw(bad_json)
   end.
 
+%% We assume the jsx representation of JSON as Erlang terms:
+%%   true/false/null: 'true' | 'false' | 'null'
+%%   number:          integer() | float()
+%%   string:          binary() | atom()
+%%   array:           [ JSON ]
+%%   object:          #{ Label => JSON, ... } |  [{ Label, JSON }] | [{}]
+%%   date string:     {{Year, Month, Day}, {Hour, Min, Sec}}
+%% where
+%%   Label:           binary() | atom() | integer()
+%%
+%% We also detect lists of printable characters (plain Erlang strings) and
+%% convert them into binaries. This use is deprecated and should be
+%% removed (for example, a json array [64] becomes <<"@">>).
+%%
+%% When normalizing, we make all strings and labels be binaries,
+%% and all objects be maps, not proplists.
+
+%% @hidden
+-spec normalize_json(jsx:json_term()) -> jsx:json_term().
+normalize_json(Json) when is_map(Json) ->
+    normalize_json_proplist(maps:to_list(Json));
+normalize_json([]) -> []; % empty array
+normalize_json([{}]) -> #{}; % special case in jsx for empty map as list
+normalize_json([{_K, _V} | _] = Json) ->
+    normalize_json_proplist(Json); % map as proplist
+normalize_json(Json) when is_list(Json) ->
+    case io_lib:printable_list(Json) of
+        true -> unicode:characters_to_binary(Json);
+        false -> normalize_json_list(Json)
+    end;
+normalize_json(true) -> true;
+normalize_json(false) -> false;
+normalize_json(null) -> null;
+normalize_json(Json) when is_atom(Json) -> atom_to_binary(Json);
```

Comment:
Can you either use `atom_to_binary/2` or just drop support for OTP21 by adjusting the minimal OTP version in `rebar.config` and removing `21` from [here](https://github.com/inaka/cowboy_swagger/blob/bf1e8a0996dbdf694710694abcd534494bb2e145/.github/workflows/ci.yml#L16)?

Reply:
changed to /2